src/app/shared/components/batchupload/batchupload.component.ts

```diff
@@ -0,0 +1,370 @@
+  @ViewChild('batchUploadNSS', {static: true}) public batchUploadModalNSS;
+
+  private configSettings: Config;
+  public modalSubscription: any;
+  public modalRef;
+  private modalElement: any;
+  public selectUpload: boolean = false;
+  public sheetNamesButtons: boolean;
+  public wb: XLSX.WorkBook
+  public data: [][];
+  public tableDisplay: boolean = false;
+  public equationData: equation[] = [];
+
+  public regions: Array<Region>;
+  public statisticGroupTypes: Array<Statisticgroup>;
+  public regressionTypes: Array<Regressiontype>;
+  public unitTypes: Array<Unittype>;
+  public regressionRegions: Array<Regressionregion>;
+  public errors: Array<Error>;
+  
+  public success = []
+  public clicked = false;
+
+  constructor(private _nssService: NSSService, 
+    private _modalService: NgbModal, 
+    private _toasterService: ToasterService, 
+    private _configService: ConfigService,
+    public _settingsservice: SettingsService
+    ) {     this.configSettings = this._configService.getConfiguration();
+    }
+
+  ngOnInit() {
+    this.modalSubscription = this._nssService.showBatchUploadModalNSS.subscribe((show: boolean) => {
+      if (show) { this.showModal(); }
+    });
+    this.modalElement = this.batchUploadModalNSS;
+
+    // Get all statistic groups, regions, unit types, regression (variable) types, errors
+    this._settingsservice.getEntities(this.configSettings.nssBaseURL + this.configSettings.statisticGrpURL).subscribe((statGroups: Array<Statisticgroup>) => {
+      this.statisticGroupTypes = statGroups;
+    });
+     this._settingsservice.getEntities(this.configSettings.nssBaseURL + this.configSettings.regionURL).subscribe((regions: Array<Region>) => {
+      this.regions = regions;
+    });
+    this._settingsservice.getEntities(this.configSettings.nssBaseURL + this.configSettings.unitsURL).subscribe((unitTypes: Array<Unittype>) => {
+      this.unitTypes = unitTypes;
+    });
+    this._settingsservice.getEntities(this.configSettings.nssBaseURL + this.configSettings.regTypeURL).subscribe((regTypes: Array<Regressiontype>)  => {
+      this.regressionTypes = regTypes;
+    });
+    this._settingsservice.getEntities(this.configSettings.nssBaseURL + this.configSettings.errorsURL).subscribe((errorTypes: Array<Error>) => {
+      this.errors = errorTypes;
+    });
+  }
+
+  public showModal(): void {
+    this.modalRef = this._modalService.open(this.modalElement, { backdrop: 'static', keyboard: false, size: 'lg', windowClass: 'modal-xl batch-upload-modal' });
+  }
+
+  public clearTable() {
+    this.sheetNamesButtons = false;
+    this.tableDisplay = false; 
+    delete(this.data);
+  }
+
+  public selectFile(event: any) {
+    const target: DataTransfer = <DataTransfer>(event.target);
+    if (target.files.length !== 1) { //Check for multiple files
+      this.clearTable();
+      this._toasterService.pop('error', 'Error', 'Cannot select multiple files');
+      return;
```

Comment:
The `multiple='false'` in the html should cover this. Since it doesn't, that indicates it's set up incorrectly. The default is to allow selection of only one file. If you remove the `multiple='false'`, it should work fine and allow only one file.